### PR TITLE
Views

### PR DESCRIPTION
**Pull request by Diegogagan2587**
# Implemented Changes

	- [x] Implemente Structure and functionalities(Ignore Login Logic.)
		- [x] Implement Reusable Components
			- [x] UserCard include
				- [x] User picture
				- [x] User name
				- [x] Number of Posts: x
			- [x] PostCard include
				- [x] Post #1
				- [x] Text
				- [x] Comments: x
				- [x] LIkes: y
			- [x] CommentsBox incldue
				- [x] Username and comment
		- [x] Implemente Users Page: /users
			- [x] Display a List of UserCard components
		- [x] Implement User page: /users/:id
			- [x] Display one UserCard for the current user
			- [x] Display BioBox component
			- [x] Display 0 up to 3 CommentsBox for current user
			- [x] Include a ***See all posts*** button
		- [x] Implemente a Page for User's Posts: /users/:id/posts
			- [x] Display one UserCard component
			- [x] Display PostCard + CommentsBox components
			- [x] Include a ***Pagination*** button
		- [x] Implement a page for only a given post from an user /users/:id/posts/:post_id
			- [x] Include a FullPost component 
			- [x] CommentsBox compoente
	- [x] Then Implemente the Design
		- [x] Add styling to all components
			- [x] users
			- [x] selected user
			- [x] posts
			- [x] for selected post
	- [x] Include all buttons and check wich buttons would work for the navigation, make them work

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,2 +1,25 @@
...
+        <ul class="posts">
+            <% @user.posts.each_with_index do |post, i| %>
+                <li>
+                  <a href="<%=user_post_path(@user,post)%>">
+                   <%= render partial: 'posts/post_card', locals: { post: post, post_number:i+1 } %>
+                  </a>
+                </li>
+            <% end %>
+        </ul>
+        <%# next button will redirect to see all post for current user %>
```
Comment (V-Blaze) [inline app/views/users/show.html.erb]: - You have done great so far :+1: But there are a few requirements you need to meet before moving to the next milestone. :+1: :100: 

- [x] In this user show page, after showing the user's details, you are meant to show only the users `three most recent posts` here. You are to do this using the `last_three_posts` method in your `user model`. I'll suggest you take this into consideration and update this. :smile: :wave:

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1 +1,23 @@
...
+       <ul>
+         <li>
+         <%= render partial: 'posts/post_card', locals: {post: post, post_number: i+1} %>
+         </li>
+         <li>
+             <%= render partial: 'application/comments_box', locals: { post: post  }%>
+         </li>
+        </ul>
+     </li>
+    <% end %>
```
Comment (V-Blaze) [inline app/views/posts/index.html.erb]: - [x] Similar to what I pointed out above, in this post index page, you are not meant to display all the comments for each post. You are meant to use the method you created earlier `five_recent_comments` and display only 5 comments on each post. I suggest you update this correctly. :+1: :100:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Diegogagan2587 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Used `rails view partials` ✔
- Design looks like the wireframe👋 
-  clean and well structured code ✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/show.html.erb (hunk 1 above)
Comment (Diegogagan2587) [reply app/views/users/show.html.erb]: @V-Blaze Thanks for pointing that out! I'll make the correction!

**Review by TuyishimireEric — APPROVED** [review]
# Status: APPROVED 🟢  
### Hi  @Diegogagan2587   👋 👋 

#### I'm @TuyishimireEric, Your code reviewer,
Good job on implementing the project requirements! 🥇 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it 🦾  
Congratulations! 🎉


## Highlight 👍 
✅ Made suggested changes 👍 

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/enus/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**